app/src/main/java/org/mozilla/fenix/home/intent/DefaultBrowserIntentProcessor.kt

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home.intent
+
+import android.content.Intent
+import androidx.navigation.NavController
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.ext.openSetDefaultBrowserOption
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.onboarding.DefaultBrowserNotificationWorker.Companion.isDefaultBrowserNotificationIntent
+
+class DefaultBrowserIntentProcessor(
+    private val activity: HomeActivity
+) : HomeIntentProcessor {
+
+    override fun process(intent: Intent, navController: NavController, out: Intent): Boolean {
+        return if (isDefaultBrowserNotificationIntent(intent)) {
+            activity.openSetDefaultBrowserOption()
+            activity.settings().shownDefaultBrowserNotification = true
```

Comment:
hand-wavy nit: I was misreading "shown" as "show" so the logic was a bit confusing at first. We have some examples of "should" but that's the opposite logic that you have here. Maybe a good past tense word could be "defaultBrowserNotificationDisplayed"/"showedDefaultBrowserNotification'?

Again, this is a pretty handy-wavy nit, so by all means, ignore it if you wish.

Reply:
Sure, I'll rename to `defaultBrowserNotificationDisplayed`.  Thanks,